tests/test_ihex.py

```diff
@@ -0,0 +1,21 @@
+from os.path import dirname
+
+import binwalk
+from nose.tools import eq_, ok_
+
+
+def test_hello_world_simple_scan():
+    '''
+    Test: Open hello-world.ihex, scan for signatures
+    verify that only one signature is returned
+    verify that the only signature returned is Intel HEX data-signature
+    '''
+    scan_result = binwalk.scan(
+        dirname(__file__) + '/input-vectors/hello-world.ihex',
+        signature=True,
+        quiet=True,
+        extract=True)  # Throws a warning for missing external extractor
+    ok_(scan_result != [])
+    eq_(len(scan_result), 1)
```

Comment:
This check verifies how many binwalk modules were executed, not how many results were reported by that module. This is also a good thing to validate, since in this case only the Signature module should have been run, it's just not the validation that I think was intended. To also verify the number of scan results, add something like:

```
eq_(len(scan_result[0]), 1)
```

Same comment for the validation check in test_srec.py.